python/mpt/src/mpt/utils.cairo

```diff
@@ -179,3 +190,77 @@ func check_branch_node(node: BranchNode) {
     }
     return ();
 }
+
+func sort_account_diff{range_check_ptr}(diff: AccountDiff) -> AccountDiff {
+    alloc_locals;
+    // Length of the array
+    let diffs_len = diff.value.len;
+    if (diffs_len == 0) {
+        return diff;
+    }
+
+    let diffs_ptr = diff.value.data;
+    // Buffer to store the sorted entries
+    let (buffer) = alloc();
+    let (sorted_indexes) = alloc();
+    %{ sort_account_diff %}
+
+    tempvar sorted_account_diffs = new AccountDiffStruct(
+        data=cast(buffer, AddressAccountNodeDiffEntry*), len=diffs_len
+    );
+    local sorted_account_diffs: AccountDiffStruct* = sorted_account_diffs;
+    local sorted_indexes: felt* = sorted_indexes;
+
+    tempvar range_check_ptr = range_check_ptr;
+    tempvar i = 0;
+    static_assert i == [ap - 1];
+
+    loop:
+    let range_check_ptr = [ap - 2];
+    let i = [ap - 1];
+    let unsorted = cast([fp - 3], AccountDiffStruct*);
+
+    // Check that the sorted array is a permutation of the unsorted array
+    // With corresponding indexes given as oracle
+    with_attr error_message("KeyError") {
+        let sorted_index = [sorted_indexes + i];
+
+        let unsorted_key_at_index = unsorted.data[sorted_index].value.key.value;
+        let sorted_key_at_index = sorted_account_diffs.data[i].value.key.value;
+
+        assert unsorted_key_at_index = sorted_key_at_index;
+        assert sorted_account_diffs.data[i].value = unsorted.data[sorted_index].value;
+    }
+
+    // Since len is >= 1 and i starts at 0,
+    // This won't underflow
+    let is_end = is_zero(unsorted.len - i - 1);
+    tempvar i = i + 1;
+    jmp end if is_end != 0;
+
+    // Check that the sorted array is ordered
+    with_attr error_message("ValueError") {
+        // If we are not at the end,
+        // We can access offset i + 1
+        let is_descending_ordered = is_le_felt(
+            sorted_account_diffs.data[i].value.key.value,
+            sorted_account_diffs.data[i - 1].value.key.value,
```

Comment:
you incremented `i` right before. so this is `i, i+1` ? misleading to shadow the `i` variable I think